### PR TITLE
(SIMP-3156) Add SSG and InSpec support to Beaker

### PR DESCRIPTION
**Pull request by trevor-vaughan**
This adds preliminary support for both SSG scanning and remediation and
InSpec scanning to Beaker for use in testing various systems and
modules.

The README has not been updated because changes may need to be made
once further integration is made.

SIMP-3156 #comment added support in beaker helpers

### CODE REVIEW

File: lib/simp/beaker_helpers/ssg.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
+module Simp::BeakerHelpers
+  # Helpers for working with the SCAP Security Guide
+  class SSG
+
+    GIT_REPO = 'https://github.com/OpenSCAP/scap-security-guide.git'
```
Comment (jeefberkey) [inline lib/simp/beaker_helpers/ssg.rb]: Can this be an ENV var?

File: lib/simp/beaker_helpers/inspec.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
+module Simp::BeakerHelpers
+
+  # Helpers for working with Inspec
+  class Inspec
+    require 'highline'
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers/inspec.rb]: Please move the require down to where it is needed.  Highline, by default, uses global objects which may screw up rubygem-simp-cli spec tests.

File: lib/simp/beaker_helpers/inspec.rb [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        fail(err_msg.join("\n"))
+      end
+    end
+
+    # Output the report
+    #
+    # @param report
+    #   The inspec results Hash
+    #
+    def write_report(report)
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers/inspec.rb]: When/how is this called by the test framework?

File: lib/simp/beaker_helpers/inspec.rb [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      File.open(@inspec_result_file + '.report', 'w') do |fh|
+        fh.puts(report[:report].uncolor)
+      end
+    end
+
+    # Process the results of an InSpec run
+    #
+    # @return [Hash] A Hash of statistics and a formatted report
+    #
+    def process_inspec_results
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers/inspec.rb]: Where is this method used?

File: lib/simp/beaker_helpers/ssg.rb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      unless File.directory?(@output_dir)
+        FileUtils.mkdir_p(@output_dir)
+      end
+
+      @result_file = "#{@sut.hostname}-ssg-#{Time.now.to_i}"
+
+      get_ssg_datastream
+    end
+
+    def target
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers/ssg.rb]: How/where is this method used?

File: lib/simp/beaker_helpers/ssg.rb [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      OS_INFO[@os][@os_rel]['ssg']['target']
+    end
+
+    def remediate(profile)
```
Comment (lnemsick-simp) [inline lib/simp/beaker_helpers/ssg.rb]: How/where is this method used?

File: lib/simp/beaker_helpers/inspec.rb (hunk 3 above)
Comment (trevor-vaughan) [inline lib/simp/beaker_helpers/inspec.rb]: The answer to the rest of these will be understood when this gets pushed and the module can use it.

File: lib/simp/beaker_helpers/inspec.rb (hunk 2 above)
Comment (trevor-vaughan) [inline lib/simp/beaker_helpers/inspec.rb]: Good call

**Comment by trevor-vaughan** [comment]
@lnemsick-simp I don't know if this is going to be the final form but I'd like to get it into the code so that we can try it in a couple of modules.

**Pull request merged**